src/main/java/it/gov/pagopa/merchant/service/RetrieveMerchantIdService.java

```diff
@@ -0,0 +1,7 @@
+package it.gov.pagopa.merchant.service;
+
+import it.gov.pagopa.merchant.dto.MerchantInfoDTO;
+
+public interface RetrieveMerchantIdService {
```

Comment:
I'd use a more generic name such as MerchantInfoService. Furthermore, verbs are generally used to name methods rather than classes or attributes